### PR TITLE
Display a more appropriate message for invalid queries

### LINKED ISSUE #489: Scheduled posts should be reflected as such in the post list.
Reported by @aerych 

When testing scheduled posts I noticed that the posts list does not have any visual indicator that a post is scheduled or already published. I thought we had this in the past but maybe not?

### LINKED ISSUE #15155: Improving Error Messaging for non-English characters in Site Creation Flow
### Expected behavior

I would expect to be told non-English characters are not supported if I'm using unsupported characters in trying to create a site. 

### Actual behavior

When trying to create a site using a language like Arabic or Chinese, we see the error `No available addresses matching your search` instead. It would be helpful if we could show a message that non-English characters are not supported in site addresses on WordPress.com. 

### Steps to reproduce the behavior

1. Navigate to the site creation flow and add a new WordPress.com site.
2. Use Chinese, Arabic, Hebrew, etc. characters to create a new site address.
3. See the error. 

IMG_10EA29E2AF13-1

##### Tested on iPhone X, iOS 13.7, WPiOS 15.9

Internal report: p4a5px-2Ez-p2#comment-11419

### PR DESCRIPTION
**Pull request by mkevins**
Fixes #15155

### Description

This PR improves the error message for invalid domain queries (e.g. when invalid characters are in the query string). The approach is very similar to the [equivalent Android PR](https://github.com/wordpress-mobile/WordPress-Android/pull/15969)

### To Test
1. Start creating a new site using the app (From My Site screen tap the ▼ icon next to the site title)
2. On the Site Picker screen tap on the ➕ menu button
3. Select **Create WordPress.com site**
4. Choose any design or tap the **SKIP** menu button
5. Type `_` or any word with non-English characters
6. **Verify** that the text message under the input is:
   > Your search includes characters not supported in WordPress.com domains. The following characters are allowed: A–Z, a–z, 0–9.

## Regression Notes
1. Potential unintended areas of impact
Nothing comes to mind

2. What I did to test those areas of impact (or what existing automated tests I relied on)
N/A

3. What automated tests I added (or what prevented me from doing so)
Automated tests for this are included in the related WordPressKit-iOS PR.

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WordPress/Classes/ViewRelated/Site Creation/WebAddress/WebAddressWizardContent.swift [hunk 1]
```diff
@@ -196,6 +196,7 @@ final class WebAddressWizardContent: CollapsableHeaderViewController {
         data = []
         lastSearchQuery = nil
         setAddressHintVisibility(isHidden: false)
+        noResultsLabel.isHidden = true
```
Comment (mkevins) [inline WordPress/Classes/ViewRelated/Site Creation/WebAddress/WebAddressWizardContent.swift:199]: This is necessary and fixes an existing (minor) issue where the no results message is not hidden after deleting the query.

Comment (mkevins) [reply WordPress/Classes/ViewRelated/Site Creation/WebAddress/WebAddressWizardContent.swift:199]: Steps to reproduce the (minor) issue on `trunk`:

1. Begin the create site flow
2. Choose a design (or tap skip)
3. Type [SPACE] - observe the no results message
4. Tap backspace

The no results message is still visible on top of the sample view

Comment (ovitrif) [reply WordPress/Classes/ViewRelated/Site Creation/WebAddress/WebAddressWizardContent.swift:199]: > This is necessary and fixes an existing (minor) issue where the no results message is not hidden after deleting the query.

Awesome find, thank you for fixing this @mkevins!
I had it on my Todo list to inform folks about that issue 🙃

**Comment by ovitrif** [comment]
Hey @mkevins! My intention when asking for your review of my Android PR [#15999](https://github.com/wordpress-mobile/WordPress-Android/pull/15999/files) was to make sure you're in the loop with our preference to align if possible on the error message for both platforms (Antonis & myself discussed about this in private).

As seen there, the final copy of the message is:

> Your search includes characters not supported in WordPress.com domains. The following characters are allowed: A–Z, a–z, 0–9.

If you find this suggestion ok, please make we use the same copy for iOS 🙇

**Comment by mkevins** [comment]
Thanks for keeping me in the loop about it Ovi!

> If you find this suggestion ok, please make we use the same copy for iOS

I intend to update this for the new messaging - actually one of the reasons it is still in draft is that I was that the final string was still "pending" on Android (also beta testing / groundskeeping :sweat_smile: ). I'll update this and make it ready for review soon. Thanks!

**Review by ovitrif — APPROVED** [review]
The code looks good, and I appreciate you fixed the error label that was showing over the domain suggestions if an "empty list state" occurred.

Thank you for doing these fixes Matthew!

- The improved error message for invalid queries works as expected ✅
- The error message is correctly hidden after a former search resulted in empty list and the next search resulted in domain suggestions ✅!

I'll try to merge this for `19.4` 🤞🏽 🚀

**Comment by brcavallaro** [comment]
Hey @mkevins!

I believe this change is going to generate a blocker for users that are with their devices in RTL languages like Arabic or Hebrew, for example. I tried to perform some tests here, and I cannot query (when using Arabic common chars), even create a new site. It seems that they are being considered in the bucket of all non-English characters.

Would you mind taking a look at that once you have a chance? Thank you!

**Comment by mkevins** [comment]
Hi @brcavallaro :wave: :smile:  Thank you for reaching out!

> I believe this change is going to generate a blocker for users that are with their devices in RTL languages like Arabic or Hebrew, for example. I tried to perform some tests here, and I cannot query (when using Arabic common chars), even create a new site. It seems that they are being considered in the bucket of all non-English characters.
> 
> Would you mind taking a look at that once you have a chance? Thank you!

To clarify, the changes in this PR were not intended to change the ability to use any particular characters in the domain query, but rather to improve the messaging about which characters are currently supported. Prior to this, when characters were entered that were considered "invalid" by the API, the screen would only display "No results", without an explanation. The intent of these changes was to present a more useful message to the user about _why_ no results are shown in such cases.

That said, I'm curious if you have encountered a difference in terms of the actual _results_ you have seen after this PR. For example, when you use the same search term on an earlier version of the app, did you see results that you no longer see? If so, could you provide the query string so I can attempt to reproduce this and investigate it? If not, please let me know as well, since this is useful feedback for the API maintainers.

**Comment by ovitrif** [comment]
> I believe this change is going to generate a blocker for users that are with their devices in RTL languages like Arabic or Hebrew, for example. I tried to perform some tests here, and I cannot query (when using Arabic common chars), even create a new site. It seems that they are being considered in the bucket of all non-English characters.

Hey @brcavallaro!
Thank you for your question. I'm not 💯 sure I understand.
Do you mean that when inputting Arabic characters (so not from the Latin alphabet) you always get the error message?

If that's the case I think it was like this before the fix, maybe check with version `19.4`.
The reason we introduced this message is to make it a bit more clear to the user what the problem is, since a while ago we got many support tickets because of this problem.

**Comment by brcavallaro** [comment]
Hi @mkevins 👋 No problem, I'm gald to help!

First of all, sorry about the delay on replying here. I've been away for a couple of days.

I just downloded the current WP published version (19.4.1) and verified that the behavior is the same as you mentioned, I mean, I see the message "No results (...)" when searching for the same term in both versions.

What intrigates me is that, it does not matter which term I use, when I am typing anything with the Arabic keyboard, nothing is ever returned - I always see the message "No results (...)". Does this mean that we have not have domains available to be returned by the APIs, or are the APIs handling these characters wrongly?

**Comment by mkevins** [comment]
Hi @brcavallaro :wave: 

No worries at all about the delay.. I have been delayed in responding as well :sweat_smile: .

Thank you for reporting your findings about the earlier version.

> What intrigates me is that, it does not matter which term I use, when I am typing anything with the Arabic keyboard, nothing is ever returned - I always see the message "No results (...)". Does this mean that we have not have domains available to be returned by the APIs, or are the APIs handling these characters wrongly?

The API itself actually returns an error message like the following:

`Domain searches only support the following characters (many accents are also allowed): A-Z, a-z, 0-9, -, ., space. You searched for the following unsupported characters: ` followed by the characters that are not currently supported. So, it seems that the API lacks support for the characters you have entered in your testing (i.e. unavailable). Fwiw, from my understanding, I believe the characters with accents which are "supported" may be "converted" to characters without accents (so the resulting domains fetched will likely not include those accented characters). This is partly why the message is simplified in this PR, and it's [Android counterpart](https://github.com/wordpress-mobile/WordPress-Android/pull/15969).

**Pull request merged**